controllers/machinedeployment_sync_test.go

```diff
@@ -397,3 +398,100 @@ func TestScaleMachineSet(t *testing.T) {
 		})
 	}
 }
+
+func newTestMachineDeployment(pds *int32, replicas, statusReplicas, updatedReplicas, availableReplicas int32, conditions clusterv1.Conditions) *clusterv1.MachineDeployment {
+	d := &clusterv1.MachineDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "progress-test",
+		},
+		Spec: clusterv1.MachineDeploymentSpec{
+			ProgressDeadlineSeconds: pds,
+			Replicas:                &replicas,
+			Strategy: &clusterv1.MachineDeploymentStrategy{
+				Type: clusterv1.RollingUpdateMachineDeploymentStrategyType,
+				RollingUpdate: &clusterv1.MachineRollingUpdateDeployment{
+					MaxUnavailable: intOrStrPtr(0),
+					MaxSurge:       intOrStrPtr(1),
+					DeletePolicy:   pointer.StringPtr("Oldest"),
+				},
+			},
+		},
+		Status: clusterv1.MachineDeploymentStatus{
+			Replicas:          statusReplicas,
+			UpdatedReplicas:   updatedReplicas,
+			AvailableReplicas: availableReplicas,
+		},
+	}
+	d.Status.Conditions = conditions
```

Comment:
qq: what does this line do? Was the idea to maybe test for modification on an existing set of conditions. If yes, seems like a good missing test case to cover.

Reply:
As we add more conditions to the machinedeployment, this will/may be useful. Please see tracking issue for additional conditions to be added: https://github.com/kubernetes-sigs/cluster-api/issues/3486

Reply:
Please inline this within the the rest of the status unless there's a reason to not, otherwise it suggests there's some nuance the reader is missing and results confusing.

```
		Status: clusterv1.MachineDeploymentStatus{
			Replicas:          statusReplicas,
			UpdatedReplicas:   updatedReplicas,
			AvailableReplicas: availableReplicas,
                         Conditions: conditions,
		},
```

>As we add more conditions to the machinedeployment, this will/may be useful.

Also as general rule please let's don't over abstract ahead of time as it makes things unnecessarily more difficult to reason about when looking at PRs, code snapshots, git history... Let's abstract when we need it if we need it.

Reply:
done.